multicluster/cmd/service-mirror/main.go

```diff
@@ -126,75 +136,106 @@ func Main(args []string) {
 			cleanupWorkers()
 		}
 	} else {
-		k8sAPI, err := k8s.NewAPI(*kubeConfigPath, "", "", []string{}, 0)
-		//TODO: Use can-i to check for required permissions
-		if err != nil {
-			log.Fatalf("Failed to initialize K8s API: %s", err)
-		}
-		linkClient := k8sAPI.DynamicClient.Resource(multicluster.LinkGVR).Namespace(*namespace)
-
 		run = func(ctx context.Context) {
-		main:
-			for {
-				// Start link watch
-				linkWatch, err := linkClient.Watch(ctx, metav1.ListOptions{})
-				if err != nil {
-					log.Fatalf("Failed to watch Link %s: %s", linkName, err)
-				}
-				results := linkWatch.ResultChan()
-
-				// Each time the link resource is updated, reload the config and restart the
-				// cluster watcher.
-				for {
-					select {
-					// ctx.Done() is a one-shot channel that will be closed once
-					// the context has been cancelled. Receiving from a closed
-					// channel yields the value immediately.
-					case <-ctx.Done():
-						// The channel will be closed by the leader elector when a
-						// lease is lost, or by a background task handling SIGTERM.
-						// Before terminating the loop, stop the workers and set
-						// them to nil to release memory.
-						cleanupWorkers()
+			// Use a small buffered channel for Link updates to avoid dropping
+			// updates if there is an update burst.
+			results := make(chan *v1alpha2.Link, 100)
```

Comment:
Would be cool as a followup to get a metric on the size of this queue and/or the number of dropped updates